fdbcli/fdbcli.actor.cpp

```diff
@@ -3656,8 +3657,34 @@ ACTOR Future<int> runCli(CLIOptions opt) {
 		[](std::string const& line, std::vector<std::string>& completions) {
 			fdbcli_comp_cmd(line, completions);
 		},
-		[](std::string const& line)->LineNoise::Hint {
-			return LineNoise::Hint();
+		[enabled=opt.cliHints](std::string const& line)->LineNoise::Hint {
+			if (!enabled) {
+				return LineNoise::Hint();
+			}
+
+			bool error = false;
+			bool partial = false;
+			std::string linecopy = line;
+			std::vector<std::vector<StringRef>> parsed = parseLine(linecopy, error, partial);
+			if (parsed.size() == 0 || parsed.back().size() == 0) return LineNoise::Hint();
+			StringRef command = parsed.back().front();
+			int finishedParameters = parsed.back().size() + error;
+
+			// We don't want the hint to flip to parse error and back, e.g. while \" is being typed.
```

Comment:
```suggestion
			// As a user is typing an escaped character, e.g. \", after the \ and before the " is typed 
			// the string will be a parse error.  Ignore this parse error to avoid flipping the hint to
			// {malformed escape sequence} and back to the original hint for the span of one character                                     
			// being entered.
```